sonic_platform_base/sfp_base.py

```diff
@@ -304,3 +304,47 @@ def set_power_override(self, power_override, power_set):
             False if not
         """
         raise NotImplementedError
+
+    def read_eeprom_devid(self, port_num, devid, offset, num_bytes):
+        """
+        read eeprom specfic bytes for a given port_num and device id
+        begining from a random offset with size as num_bytes
+
+        Args:
+             port_num :
+                     Integer, index of physical port from where to read the bytes
+             devid :
+                     Integer, device id of the i2c bus required for reading for eeprom addr space
+             offset :
+                     Integer, the offset from which the read transaction will start
+             num_bytes:
+                     Integer, the number of bytes to be read
+
+        Returns:
+            a list , raw sequence of bytes which are read from the offset of size num_bytes
```

Comment:
Should this return a list or a byte array? We also must keep in mind we are transitioning to Python 3, where strings are handled differently. We should plan ahead.

Reply:
I will change this to a byte array .